### ISSUE TITLE
Bug Take Off, switch TOGA => CLB ...

### ISSUE THREAD

**Issue by vincenzo34** [flybywiresim/a32nx#5587]
### Aircraft Version

Development

### Build info

```json
{
  "built": "2021-08-07T04:01:01+00:00",
  "ref": "master",
  "sha": "10169b43abb8568044bab599783275b34f2bca70",
  "actor": "tracernz",
  "event_name": "manual"
}
```

### Describe the bug

Hello,

I am having a problem since upgrading to version 1.18.15.0 of MSFS and DEV version of FlybyWire.

During Take-off, it is impossible to switch from TOGA => CLB ... for a while ... there are flashing of HDG, VS and others ... 
It is impossible to switch to HDG or OB_climb step...
After a certain altitude the problem stops and everything works ...

Video link: https://streamable.com/3lqaxa

Thx 
Vincent

### Expected behavior

Normally, switching to TOGA => CLB does not stop a flashing problem ...

### Steps to reproduce

Cold and Dark.
Engine Start.

All the times.

**Comment by aguther** [comment]
After an update of the A32NX please load your flight and after pressing "FLY" reload it. It seems something is not getting initialized correctly and it's unclear what it is yet.

**Comment by edmondchaimy** [comment]
I've also experienced the same bug.

**Comment by aguther** [comment]
Did it help what I proposed?

**Comment by aguther** [comment]
…and did you enter a V2? Usually you should be in SRS and not V/S on take-off…

**Comment by vincenzo34** [comment]
Hi , no but : 

I noticed that when I correctly prepare for the flight by completely filling out the MCDU (FPLN, INIT, PERF ...), I don't have a problem ... on the other hand if I want to do a flight test without touching at the FMC, the problem exists at take off !

Not putting the V1 VR and V2 speeds could be the cause?

**Comment by vincenzo34** [comment]
I think you can reproduce the problem by starting a flight without touching the MCDU ...

**Comment by aguther** [comment]
Thank you for the info, I will try that (do a flight without touching MCDU). It worked in the past for sure, but this info hopefully guides me to the root cause and I hope I can reproduce the issue with it.

As background information:
Entering the V2 is quite important, because it decides if SRS engages or not (if you have FD on). So not entering a V2 and taking-off with FD on is a non-standard operation (SOP). There is a SOP with FDs off where you don‘t get SRS obviously, but even that is rarely used.

**Comment by vincenzo34** [comment]
Thank you for the info aguther ;)
Let me see if you want another info ;)

**Comment by aguther** [comment]
Unfortunately I cannot reproduce it up to now. We try further.

**Comment by vincenzo34** [comment]
Ok test this morning and in effect impossible to reproduce the bug .... strange ...

**Comment by aguther** [comment]
Did you update the plane last time right before it occurred? Did you start cold and dark or on the runway?

**Comment by vincenzo34** [comment]
Yes i have update the plane yesterday (but i have the last version before) ... start on the runway ...

**Comment by vincenzo34** [comment]
ok new test with cold and dark ... and the problem apear !

**Comment by aguther** [comment]
But you did not update the plane before?

What steps did you take from C&D? What did you enter on the MCDU? What was the state of the FCU and the FMA on take-off?

**Comment by vincenzo34** [comment]
Tablet On
Bat 1 & 2 => On
External Power => On
Fuel pomp => On
Lights => All On
Seat Belt & smoking => On
Emergency => Arm
Cockpit => Close
APU => Start
HDG => Runway Heading
Altitude => 30000
IAS => 250
When apu start => Apu Bleed On , External Power Off
Push back and Engine 2 start => Engine 1 start
Apu Bleed => Off
Pilot Heat => On
Flaps => 1
Spoilers => Arm
Auto Breack => Max
Wheater radar => On
Transponder => On
Taxi to runway
Take Off

NO touch MCDU !

I back later with screenshot FCU and FMA.

**Comment by aguther** [comment]
Ok, will try again the same. In my test it worked from C&D.

Can you maybe provide a FDR file from a flight where the issue occurred?

Here is a description where to find the work folder with the files:
https://docs.flybywiresim.com/start/autopilot-fbw/

**Comment by NUTTTTTTTT** [comment]
> ok new test with cold and dark ... and the problem apear !

Had you entered V1, VR and V2 in this test?
I've seen this crop up in #Support on discord, and V speeds weren't set in that case either.

**Comment by aguther** [comment]
As he said he didn’t touch the MCDU at all. So no V-Speeds.

The strange thing is I tested it exactly that way and didn’t get the error.

**Comment by vincenzo34** [comment]
I made a video that I am uplaoding I give you the link as soon as it is online !

**Comment by vincenzo34** [comment]
https://www.youtube.com/watch?v=9lNtMkVs_e8

**Comment by aguther** [comment]
I tried again: I still cannot reproduce the issue.

Can you please provide an FDR file from a flight where it happened?

**Comment by vincenzo34** [comment]
Hello,

https://easyupload.io/akxsx4

Thx
Vincent

**Comment by aguther** [comment]
@vincenzo34 Thank you for sharing. It indeed helped to get one step further. The reason why there are issues is that the flight phase is not correctly updated, it stays in PREFLIGHT (=0):
![image](https://user-images.githubusercontent.com/6629914/129944024-10629042-0214-47c0-ab8a-b40a47bd0184.png)

You can see that when this is updated and is correct it works:
![image](https://user-images.githubusercontent.com/6629914/129944199-8b2a3fff-c78a-4546-902d-e4da6407cc7d.png)

So now the big question is why this is happening...

**Comment by tracernz** [comment]
Need V2 to enter takeoff phase.
https://github.com/flybywiresim/a32nx/blob/b7076762217e0fa89881c4425816500a5d9f5b6a/flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/FMGC/A32NX_FlightPhaseManager.js#L4

**Comment by aguther** [comment]
That’s in my case not triggering the issue, there must be something on top.

**Comment by tracernz** [comment]
> That’s in my case not triggering the issue, there must be something on top.

The other way is the RA rounded to the nearest 100 feet is different to the indicated altitude rounded to the nearest 100 feet (that could be 1 foot difference, or >= 50 feet difference depending on your actual altitude...) and RA > 1.5 (airborne?)?
```js
     const ra = Simplane.getAltitudeAboveGround(); 
     return Math.round(ra / 100) !== Math.round(Simplane.getAltitude() / 100) && ra > 1.5
```
That sort of means that if you're off the ground and your QNH is wrong??

Note:
Simplane.getAltitude returns the simvar `INDICATED ALTITUDE` in feet.
Simplane.getAltitudeAboveGround returns the simvar `PLANE ALT ABOVE GROUND MINUS CG` in feet if positive, else 0.

**Comment by MisterChocker** [comment]
> Need V2 to enter takeoff phase.

This there is also a side condition which allows for V2 to be undefined.

> That sort of means that if you're off the ground and your runway is not at sea level??

That little math thingy there has been implemented to detect when the terrain is actually loaded to prevent false phase changes due to the sim still loading.

**Comment by MisterChocker** [comment]
@vincenzo34 at which airport did this happen? Does it happen at different airports also?

**Comment by vincenzo34** [comment]
Yes , LFMT, LEBL ...

**Comment by tracernz** [comment]
> LFMT

17 feet elevation

> LEBL

14 feet elevation

` Math.round(ra / 100) !== Math.round(Simplane.getAltitude() / 100)` will not work here. :(

**Comment by MisterChocker** [comment]
@tracernz we could also remove the rounding (or at least round to a full number (integer)) instead of a FL and try if this solves the issue but is still practicable.

**Comment by Peyto79** [comment]
I just made a test flight after reading this ticket. Indeed, I was always taking off in Bremen. The airport is close to sea level. 
This time i faked the QNH to make the Airbus believe it would be at 300ft above sea level. —> No issue! 
(I did not touch the MCDU before takeoff, no V2 entered)
